src/component/admin/projectRoles/ProjectRoles/ProjectRoleList/ProjectRoleList.tsx

```diff
@@ -54,60 +63,199 @@ const ProjectRoleList = () => {
-                    // @ts-expect-error
-                    setCurrentRole={setCurrentRole}
-                    setDelDialog={setDelDialog}
-                />
-            );
-        });
-    };
+    const columns = useMemo(
+        () => [
+            {
+                id: 'Icon',
+                Cell: () => (
+                    <Box
+                        data-loading
+                        sx={{
+                            pl: 2,
+                            pr: 1,
+                            display: 'flex',
+                            alignItems: 'center',
+                        }}
+                    >
+                        <SupervisedUserCircle color="disabled" />
+                    </Box>
+                ),
+            },
+            {
+                Header: 'Project role',
+                accessor: 'name',
+            },
+            {
+                Header: 'Description',
+                accessor: 'description',
+                width: '90%',
+            },
+            {
+                Header: 'Actions',
+                id: 'Actions',
+                align: 'center',
+                Cell: ({
+                    row: {
+                        original: { id, type, name, description },
+                    },
+                }: any) => (
+                    <Box
+                        sx={{ display: 'flex', justifyContent: 'center' }}
+                        data-loading
+                    >
+                        <PermissionIconButton
+                            data-loading
+                            disabled={type === BUILTIN_ROLE_TYPE}
+                            onClick={() => {
+                                navigate(`/admin/roles/${id}/edit`);
+                            }}
+                            permission={ADMIN}
+                            tooltipProps={{ title: 'Edit role' }}
+                        >
+                            <Edit />
+                        </PermissionIconButton>
+                        <PermissionIconButton
+                            data-loading
+                            disabled={type === BUILTIN_ROLE_TYPE}
+                            onClick={() => {
+                                setCurrentRole({
+                                    id,
+                                    name,
+                                    description,
+                                } as IProjectRole);
+                                setDelDialog(true);
+                            }}
+                            permission={ADMIN}
+                            tooltipProps={{ title: 'Remove role' }}
+                        >
+                            <Delete />
+                        </PermissionIconButton>
+                    </Box>
+                ),
+                width: 100,
+                disableSortBy: true,
+            },
+        ],
+        []
+    );
 
-    if (!roles) return null;
+    const initialState = useMemo(
```

Comment:
Can also be `const [initialState] = useState(...)` (no setter), but this is fine.